### ISSUE TITLE
Global variables do not show up between flowcharts in variable drawer

### ISSUE THREAD

**Issue by bentaller-lion** [snozbot/fungus#905]
**Describe the bug**

Global variables should show up in different flowcharts across the project.  They currently do not.

**To Reproduce**

1.) Make a global variable in a flowchart.
2.) Make another flowchart
3.) Observe the global variable does not exist.
4.) Also observe that if you create an If or Set block this global variable also does not exist.

**Expected behavior**

Global variable should be visible everywhere

**Screenshots**

**Versions & Platform (please complete the following information):**
 - Unity version [e.g. 2019.2.20f1 or 2017.4.25]
 - Fungus Version[e.g. v.3.12.0] This can be found in your Docs/ChangeLog or via Preferences->Fungus
 - Editor OS: [e.g. Win 10  or OSX Mavericks]
 - Build Platform [e.g. WebGL, Windows IL2CPP, iOS, Android]

**Additional context**

**Comment by breadnone** [comment]
@bentaller-lion  I've never used Global variable in Fungus, as most of mine are just public(yes, it works across flowcharts and scenes as well). If I'm not mistaken you must create a variable the same name with the Global variable on 1st flowchart on your 2nd Flowchart.

**Comment by danac78** [comment]
@breadnone You are correct. Therefore, The Flowcharts are working as expected, so not a bug.

**Comment by danac78** [comment]
For the if-then, it only works for locally available variables (what is set in each flowchart). If you create the global variable in the flowchart with the same name as in the other one, it will show up in if-then.

**Comment by bentaller-lion** [comment]
Well, if that works that's cool, but it's certainly not expected behavior.

Expected behavior is that a global defined in one chart just shows up
everywhere, and I couldn't find any documentation anywhere describing the
way it actually works.

The reason this is so important is you may have a team of people building a
game and it's really hard to coordinate what variable names are global and
what variable names are local to prevent collisions.  You can of course use
a naming convention like global_ but this seems more like a bandaid
workaround than a fix.

Anyway thanks for letting me know :)

**Comment by danac78** [comment]
@bentaller-lion  The reason I called it expected behavior because it is explained here:

https://github.com/snozbot/fungus/wiki

This wiki has a lot of useful info on how the different functions of Fungus work because it is not always apparent. :)

**Comment by danac78** [comment]
More specifically, 

https://github.com/snozbot/fungus/wiki/variables

It is under Variable scope where it describes how Private, Public and Global variables work in Fungus.

**Comment by danac78** [comment]
At this point, I recommend one of two things.

A. Close it because it is not a bug. The expected behavior per how documentation described how globals work in Fungus still working.
B. Recategorize it as a feature request and change the information to how you would like Globals to work. Currently, the flowcharts will copy information from variables that have the same name and set to global between flowcharts. If you have an idea that will allow the Flowchart to be able to do so, that would be helpful.
